File: Main/Globals/Model/InvoiceLineItem.js
class InvoiceLineItem
{
    #label;
    #amountMinor;
    #currency;

    constructor({label = '', amountMinor = 0, currency = 'INR'} = {})
    {
        this.setLabel(label);
        this.setAmountMinor(amountMinor);
        this.setCurrency(currency);
    }

    getLabel()
    {
        return this.#label;
    }

    setLabel(value)
    {
        if (value !== null)
        {
            value = String(value);
            if (value.length > 256)
            {
                value = value.slice(0, 256);
            }
        }
        this.#label = value;
    }

    getAmountMinor()
    {
        return this.#amountMinor;
    }

    setAmountMinor(value)
    {
        if (value !== null)
        {
            value = parseInt(value, 10);
            if (isNaN(value))
            {
                value = 0;
            }
        }
        this.#amountMinor = value;
    }

    getCurrency()
    {
        return this.#currency;
    }

    setCurrency(value)
    {
        if (value !== null)
        {
            value = String(value);
            if (value.length > 8)
            {
                value = value.slice(0, 8);
            }
        }
        this.#currency = value;
    }

    toJson()
    {
        return {
            label: this.getLabel(),
            amountMinor: this.getAmountMinor(),
            currency: this.getCurrency(),
        };
    }

    static fromJson(json)
    {
        const instance = new InvoiceLineItem({
            label: json.label ?? null,
            amountMinor: json.amountMinor ?? null,
            currency: json.currency ?? null
        });
        return instance;
    }
}

export default InvoiceLineItem;

File: Agent/Globals/Model/InvoiceLineItem.py

class InvoiceLineItem:
    def __init__(self, label: str = '', amount_minor: int = 0, currency: str = 'INR') -> None:
        self.set_label(label)
        self.set_amount_minor(amount_minor)
        self.set_currency(currency)

    def get_label(self) -> str:
        return self.__label

    def set_label(self, value: str) -> None:
        if value is not None:
            value = str(value)
            if len(value) > 256:
                value = value[:256]
        self.__label = value

    def get_amount_minor(self) -> int:
        return self.__amount_minor

    def set_amount_minor(self, value: int) -> None:
        if value is not None:
            try:
                value = int(value)
            except (ValueError, TypeError):
                value = 0
        self.__amount_minor = value

    def get_currency(self) -> str:
        return self.__currency

    def set_currency(self, value: str) -> None:
        if value is not None:
            value = str(value)
            if len(value) > 8:
                value = value[:8]
        self.__currency = value

    def to_json(self) -> dict:
        return {
            'label': self.get_label(),
            'amountMinor': self.get_amount_minor(),
            'currency': self.get_currency(),
        }

    @classmethod
    def from_json(cls, data: dict) -> 'InvoiceLineItem':
        instance = cls(
            label=data.get('label'),
            amount_minor=data.get('amountMinor'),
            currency=data.get('currency')
        )
        return instance
